client/state/selectors/is-eligible-for-checkout-to-checklist.js

```diff
@@ -10,26 +10,31 @@ import {
 	hasTransferProduct,
 	hasPlan,
 	hasConciergeSession,
-	hasEcommercePlan,
 } from 'lib/cart-values/cart-items';
 import isEligibleForDotcomChecklist from './is-eligible-for-dotcom-checklist';
+import { retrieveSignupDestination } from 'signup/utils';
 
 /**
  * @param {Object} state Global state tree
  * @param {Number} siteId Site ID
  * @param {Object} cart object
+ * @param {String} destination The final destination URL after signup completes
  * @return {Boolean} True if current user is able to see the checklist after checkout
  */
-export default function isEligibleForCheckoutToChecklist( state, siteId, cart ) {
+export default function isEligibleForSignupDestination( state, siteId, cart ) {
 	if (
 		hasDomainMapping( cart ) ||
 		hasDomainRegistration( cart ) ||
 		hasTransferProduct( cart ) ||
-		( ! hasPlan( cart ) && ! hasConciergeSession( cart ) ) ||
-		hasEcommercePlan( cart )
```

Comment:
This function will evaluate eligibility even for the eCommerce plan. An eligible plan will be redirected to the flow destination (thank you page for the eCommerce and ecommerce-onboarding flows).